tpgtools/resource.go

```diff
@@ -107,6 +107,10 @@ type Resource struct {
 	// after the Creation call for a resource.
 	PostCreateFunction *string
 
+	// ValidationFunction is the name of a function that's called immediately
+	// after the Creation call for a resource.
+	ValidationFunction *string
```

Comment:
this would also need to be accessed via template code in https://github.com/GoogleCloudPlatform/magic-modules/blob/main/tpgtools/templates/resource.go.tmpl to affect DCL resource generation

Reply:
added